gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/apim/core/documentation/usecase/ApiCreateDocumentationPageUsecaseTest.java

```diff
@@ -0,0 +1,296 @@
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import fixtures.core.model.AuditInfoFixtures;
+import inmemory.*;
+import io.gravitee.apim.core.audit.domain_service.AuditDomainService;
+import io.gravitee.apim.core.audit.model.AuditInfo;
+import io.gravitee.apim.core.documentation.domain_service.CreateApiDocumentationDomainService;
+import io.gravitee.apim.core.documentation.model.Page;
+import io.gravitee.apim.core.sanitizer.HtmlSanitizer;
+import io.gravitee.apim.infra.json.jackson.JacksonJsonDiffProcessor;
+import io.gravitee.apim.infra.sanitizer.HtmlSanitizerImpl;
+import io.gravitee.rest.api.service.common.UuidString;
+import io.gravitee.rest.api.service.exceptions.PageContentUnsafeException;
+import java.util.List;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.*;
+
+class ApiCreateDocumentationPageUsecaseTest {
+
+    private static final String ORGANIZATION_ID = "organization-id";
+    private static final String ENVIRONMENT_ID = "environment-id";
+    private static final String USER_ID = "user-id";
+    private static final AuditInfo AUDIT_INFO = AuditInfoFixtures.anAuditInfo(ORGANIZATION_ID, ENVIRONMENT_ID, USER_ID);
+    private static final String API_ID = "api-id";
+    private static final String PARENT_ID = "parent-id";
+    private static final String PAGE_ID = "page-id";
+
+    private final PageQueryServiceInMemory pageQueryService = new PageQueryServiceInMemory();
+    private final PageCrudServiceInMemory pageCrudService = new PageCrudServiceInMemory();
+    private final PageRevisionCrudServiceInMemory pageRevisionCrudService = new PageRevisionCrudServiceInMemory();
+    AuditCrudServiceInMemory auditCrudService = new AuditCrudServiceInMemory();
+    UserCrudServiceInMemory userCrudService = new UserCrudServiceInMemory();
+    HtmlSanitizer htmlSanitizer = new HtmlSanitizerImpl();
+    CreateApiDocumentationDomainService createApiDocumentationDomainService;
+    ApiCreateDocumentationPageUsecase apiCreateDocumentationPageUsecase;
+
+    @BeforeEach
+    void setUp() {
+        UuidString.overrideGenerator(() -> PAGE_ID);
+
+        createApiDocumentationDomainService =
+            new CreateApiDocumentationDomainService(
+                pageQueryService,
+                pageCrudService,
+                pageRevisionCrudService,
+                new AuditDomainService(auditCrudService, userCrudService, new JacksonJsonDiffProcessor())
+            );
+        apiCreateDocumentationPageUsecase =
+            new ApiCreateDocumentationPageUsecase(createApiDocumentationDomainService, pageCrudService, htmlSanitizer);
+    }
+
+    @AfterEach
+    void tearDown() {
+        Stream
+            .of(pageQueryService, pageCrudService, pageRevisionCrudService, auditCrudService, userCrudService)
+            .forEach(InMemoryAlternative::reset);
+    }
+
+    @AfterAll
+    static void afterAll() {
+        UuidString.reset();
+    }
+
+    @Nested
+    class CreateMarkdownTests {
```

Comment:
It is just a suggestion but I would reorder the tests case:

- should_create_markdown_page
- should_change_homepage_to_new_page
- should_not_add_missing_parent
- should_throw_error_if_markdown_unsafe

I would also add a test case checking the audit and pagerevision